sdk/go/common/resource/plugin/plugin.go

```diff
@@ -185,7 +185,8 @@ func dialPlugin[T any](
 	// TODO[pulumi/pulumi#337]: in theory, this should be unnecessary.  gRPC's default WaitForReady behavior
 	//     should auto-retry appropriately.  On Linux, however, we are observing different behavior.  In the meantime
 	//     while this bug exists, we'll simply do a bit of waiting of our own up front.
-	timeout, _ := context.WithTimeout(context.Background(), pluginRPCConnectionTimeout)
+	timeout, cancel := context.WithTimeout(context.Background(), pluginRPCConnectionTimeout)
```

Comment:
The linter didn't check `x/net/context`, this was getting flagged for a lost cancel, same below